torch/onnx/symbolic_helper.py

```diff
@@ -374,96 +419,104 @@ def _if_scalar_type_as(g: _C.Graph, self, tensor):
     return self
 
 
-def _is_none(x):
+def _is_none(x: _C.Value) -> bool:
     return x.node().mustBeNone()
 
 
-def _is_value(x):
+def _is_value(x: Any) -> bool:
     return isinstance(x, _C.Value)
 
 
-def _is_constant(value):
-    return not _is_value(value) or value.node().kind() in (
+def _is_constant(value: Any) -> bool:
+    return not _is_value(value) or value.node().kind() in {
         "onnx::Constant",
         "prim::Constant",
-    )
+    }
 
 
-def _is_tensor(x):
+def _is_tensor(x: _C.Value) -> bool:
     return x.type().isSubtypeOf(_C.TensorType.get())
 
 
-def _is_list(x):
+def _is_list(x: _C.Value) -> bool:
     return isinstance(x.type(), _C.ListType)
 
 
-def _is_tensor_list(x):
-    return _is_list(x) and isinstance(x.type().getElementType(), _C.TensorType)
+def _is_tensor_list(x: _C.Value) -> bool:
+    if not _is_list(x):
+        return False
+    x_type = x.type()
+    x_type = typing.cast(_C.ListType, x_type)
```

Comment:
x.type() is annotated to be JitType or something which is the base type for ListType. I haven't found a good way to annotate this polymorphic return types. cast is needed to tell mypy that it is ListType and thus carry more methods we can use on this object. I think mypy is still going to complain about `x_type: _C.ListType = x.type()` since it would be an implicit cast ([which can only happen for Any)](https://adamj.eu/tech/2021/07/06/python-type-hints-how-to-use-typing-cast/). I suppose mypy doesn't understand `_is_list`? typing.cast is a no-op in run time, but I agree it can be noisy

Reply:
I wonder if this works
```python
def _as_list_type(type: _C.JitType) -> Optional[_C.ListType]:
    if isinstance(type, _C.ListType):
        return type
    return None
```
```python
x_type = _as_list_type(x.type())
if not x_type:
    return False
element_type = str(x_type.getElementType())
```
If it does it would appear `_as_list_type` is better than `_is_list` for mypy's sake.

Reply:
I updated to use `_as_list_type`. ptal